### PR TITLE
sbcl_2_1_8: init at 2.1.8

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12372,6 +12372,7 @@ with pkgs;
   sbcl_2_0_9 = callPackage ../development/compilers/sbcl/2.0.9.nix {};
   sbcl_2_1_1 = callPackage ../development/compilers/sbcl/2.1.1.nix {};
   sbcl_2_1_2 = callPackage ../development/compilers/sbcl/2.1.2.nix {};
+  sbcl_2_1_8 = callPackage ../development/compilers/sbcl/2.1.8.nix {};
   sbcl = sbcl_2_1_2;
```
Comment (marsam) [inline pkgs/top-level/all-packages.nix:12376]: I think you also need to set it as the default sbcl:
```suggestion
  sbcl = sbcl_2_1_8;
```

Comment (sternenseemann) [reply pkgs/top-level/all-packages.nix:12376]: Haven't done any sort of testing on `lispPackages` with that sbcl version. As far as I know SBCL 2.1.8 changes some warnings (likely the versions before as well) which tends to break compilation for some packages.

I've done tests with out of tree packages though and those seemed to cause no issues.

**Comment by 7c6f434c** [comment]
OK, let's get this now and play with versions on next Quicklisp bump when rechecking `lispPackages` would be needed anyway…

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12372,6 +12372,7 @@ with pkgs;
   sbcl_2_0_9 = callPackage ../development/compilers/sbcl/2.0.9.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:12372]: Can we remove some of the older versions?

**Pull request merged**